src/libsystemd-network/dhcp-static-lease.c

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "dhcp-internal.h"
+#include "dhcp-server-internal.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+static sd_dhcp_static_lease* sd_dhcp_static_lease_free(sd_dhcp_static_lease *i) {
+        free(i->client_id.data);
+
+        return mfree(i);
+}
+
+static int dhcp_static_lease_new(DHCPClientId *client_id, const struct in_addr *address, sd_dhcp_static_lease **ret) {
+        assert_return(ret, -EINVAL);
+        assert_return(address, -EINVAL);
+
+        sd_dhcp_static_lease *p = new(sd_dhcp_static_lease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        *p = (sd_dhcp_static_lease) {
+                .n_ref = 1,
+                .client_id = *client_id,
+                .address = address->s_addr
+        };
+
+        *ret = p;
+        return 0;
+}
+
+int sd_dhcp_static_lease_new_by_mac(const uint8_t *mac_addr, size_t mac_addr_size, const struct in_addr *address, sd_dhcp_static_lease **ret){
+        assert_return(mac_addr, -EINVAL);
+        assert_return(mac_addr_size, -EINVAL);
+        DHCPClientId c;
```

Comment:
maybe initialize the structure here, we are big fans of structured initialization, i.e.: 

```c
DHCPClientId c = {
        .length = ETH_ALEN + 1;
};
```

this way the structure is fully initialized always, which is a good thing